greenpithumb/pump.py

```diff
@@ -83,9 +83,11 @@ def pump_if_needed(self, moisture):
             The amount of water pumped, in mL.
         """
         if self._pump_scheduler.is_running_pump_allowed():
-            if moisture < self._moisture_threshold:
+            if moisture < self._moisture_threshold or (
+                    self._pump_scheduler.pump_interval_exceeded()):
```

Comment:
(optional): The logic's getting a bit complex here. Consider splitting out something like

```python
def _should_pump(self, moisture):
  if not self._pump_schedulre.is_running_pump_allowed():
    return False
  return (moisture < self._moisture_threshold) or self._pump_scheduler.pump_interval_exceeded()
```